dts/bindings/pinctrl/nxp,mcux-rt-pinctrl.yaml

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) 2021 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  The node has the 'pinctrl' node label set in MCUX RT SoC's devicetree. These
+  node can be autogenerated using the MCUXpresso config tools combined with
+  the parse-rt-mex.py script in the NXP HAL.
+
+
+compatible: "nxp,mcux-rt-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        property-allowlist:
+          - drive-open-drain
+
+properties:
+    reg:
+      required: true
+
+child-binding:
+  description: |
+    MCUX RT pin controller pin configuration group.
+  child-binding:
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          An array of values defining the configuration register for a given
+          pinmux mode. See fsl_iomuxc.h in the NXP HAL for a list.
+      nxp,software_input_on:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+        description: |
+          00 Pin input follows peripheral
+          01 Pin is forced to input
+      nxp,hysteresis_enable:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+        description: |
+          00 Hyst. Disable Field
+          01 Hyst. Enable Field
+      nxp,pull_up_down_config:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+          - 2
+          - 3
+        description: |
+          00 PUS_0_100K_Ohm_Pull_Down — 100K Ohm Pull Down
+          01 PUS_1_47K_Ohm_Pull_Up — 47K Ohm Pull Up
+          10 PUS_2_100K_Ohm_Pull_Up — 100K Ohm Pull Up
+          11 PUS_3_22K_Ohm_Pull_Up — 22K Ohm Pull Up
+      nxp,pull_keeper_enable:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+        description: |
+          0 PKE_0_Pull_Keeper_Disabled — Pull/Keeper Disabled
+          1 PKE_1_Pull_Keeper_Enabled — Pull/Keeper Enabled
+      nxp,pull_keeper_select:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+        description: |
+          0 PUE_0_Keeper — Keeper
+          1 PUE_1_Pull — Pull
+      nxp,drive_strength:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+          - 2
+          - 3
+          - 4
+          - 5
+          - 6
+          - 7
+        description: |
+          000 DSE_0_output_driver_disabled_ — output driver disabled
+          001 DSE_1_R0_150_Ohm___3_3V__260_Ohm_1_8V_ — R0(150 Ohm @ 3.3V, 260 Ohm@1.8V)
+          010 DSE_2_R0_2 — R0/2
+          011 DSE_3_R0_3 — R0/3
+          100 DSE_4_R0_4 — R0/4
+          101 DSE_5_R0_5 — R0/5
+          110 DSE_6_R0_6 — R0/6
+          111 DSE_7_R0_7 — R0/7
+      nxp,speed:
```

Comment:
ST uses `slew-rate` for that, even though you already seem to have a separate slew rate property.

Reply:
I guess this can not be aligned as there is dedicatd HW setting differences.